src/requestVariableHoverProvider.ts

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+import { HoverProvider, Hover, MarkedString, TextDocument, CancellationToken, Position, Range, TextLine } from 'vscode';
+
+import { VariableProcessor } from './variableProcessor';
+import { VariableUtility } from './variableUtility';
+import { RequestVariableCacheValueProcessor } from "./requestVariableCacheValueProcessor";
+
+export class RequestVariableHoverProvider implements HoverProvider {
+
+    public async provideHover(document: TextDocument, position: Position, token: CancellationToken): Promise<Hover> {
+        if (!VariableUtility.isRequestVariableReference(document, position)) {
+            return;
+        }
+
+        // Lookup word + optional array path
+        const wordRange = document.getWordRangeAtPosition(position, /\w+(\[\d+\])*/);
+        let lineRange = document.lineAt(position);
+
+        const fullPath = this.getRequestVariableHoverPath(wordRange, lineRange, position);
+
+        const fileRequestVariables = VariableProcessor.getRequestVariablesInFile(document);
+        for (let [variableName, variableValue] of fileRequestVariables) {
+            let regex = new RegExp(`(${variableName})($|\.|\[\d+\])`);
+            if (regex.test(fullPath)) {
+                const value = await RequestVariableCacheValueProcessor.getValueAtPath(variableValue, fullPath);
+
+                let contents: MarkedString[] = [JSON.stringify(value), { language: 'http', value: `Request Variable ${variableName}` }];
+                return new Hover(contents, wordRange);
+            }
+        }
+
+        let contents: MarkedString[] = [{ language: 'http', value: `Warning: Request Variable ${fullPath} is not loaded in memory` }];
+        return new Hover(contents, wordRange);
+    }
+
+    private getRequestVariableHoverPath(wordRange: Range, lineRange: TextLine, position: Position) {
+        let index = position.character - 1;
```

Comment:
If we don't consider the case that space is allowed between variable name and `{{...}}`, can we simply remove the `for` loop